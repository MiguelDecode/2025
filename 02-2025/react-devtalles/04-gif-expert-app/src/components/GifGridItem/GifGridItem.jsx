import PropTypes from "prop-types";

const GifGridItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

export default GifGridItem;

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};