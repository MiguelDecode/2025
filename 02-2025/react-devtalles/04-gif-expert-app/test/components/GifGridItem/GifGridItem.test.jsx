import GifGridItem from "../../../src/components/GifGridItem/GifGridItem";
import { render, screen } from "@testing-library/react";

describe("Tests on component <GifGridItem/>", () => {
  const testTitle = "Test title";
  const testUrl = "http://localhost/testing";

  test("should match with the snapshoot", () => {
    const { container } = render(
      <GifGridItem title={testTitle} url={testUrl} />
    );

    expect(container).toMatchSnapshot();
  });

  test("should title and url are required", () => {
    render(<GifGridItem title={testTitle} url={testUrl} />);

    // screen.debug();

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(testUrl);
    expect(alt).toBe(testTitle);
  });

  test("should show the title in the component", () => {
    render(<GifGridItem title={testTitle} url={testUrl} />);

    expect(screen.getByText(testTitle)).toBeTruthy;
  });
});

