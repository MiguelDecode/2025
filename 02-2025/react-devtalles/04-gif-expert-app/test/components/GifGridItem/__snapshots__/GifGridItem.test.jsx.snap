// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on component <GifGridItem/> should match with the snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test title"
      src="http://localhost/testing"
    />
    <p>
      Test title
    </p>
  </div>
</div>
`;
